core/kazoo_config/src/kapps_config_util.erl

```diff
@@ -0,0 +1,86 @@
+-module(kapps_config_util).
+-include_lib("kazoo/include/kz_types.hrl").
+-export([
+         get_config/2
+        ,get_reseller_config/2
+        ,load_config_from_account/2
+        ,account_schema_name/1
+        ,system_schema_name/1
+        ,account_schema/1
+        ,system_schema/1
+        ]).
+
+-spec doc_id(ne_binary()) -> ne_binary().
+doc_id(Config) -> kapps_account_config:config_doc_id(Config).
+
+-spec get_config(ne_binary(), ne_binary()) -> kz_json:object().
+get_config(Account, Config) ->
+    Programm = [
+                fun load_config_from_account/2
+               ,fun load_config_from_reseller/2
+               ,fun load_config_from_system/2
+               ,fun load_default_config/2
+               ],
+    Schema = account_schema(Config),
+    kz_json_schema:filter(get_config(Account, Config, Programm), Schema).
+
+-spec get_config(ne_binary(), ne_binary(), [fun()]) -> kz_json:object().
+get_config(Account, Config, Programm) ->
+    Confs = [ maybe_new(P(Account, Config)) || P <- Programm ],
+    kz_json:merge_recursive(lists:reverse(Confs)).
+
+-spec get_reseller_config(ne_binary(), ne_binary()) -> kz_json:object().
+get_reseller_config(Account, Config) ->
+    Programm = [
+                fun load_config_from_reseller/2
+               ,fun load_config_from_system/2
+               ,fun load_default_config/2
+               ],
+    Schema = account_schema(Config),
+    kz_json_schema:filter(get_config(Account, Config, Programm), Schema).
+
+-spec load_config_from_account(api_binary(), ne_binary()) -> {ok, kz_json:object()} | {error, any()}.
+load_config_from_account(undefined, _Config) -> kz_json:new();
+load_config_from_account(Account, Config) ->
+    AccountDb = kz_util:format_account_id(Account, encoded),
+    kz_datamgr:open_cache_doc(AccountDb, doc_id(Config), [{cache_failures, [not_found]}]).
+
+-spec load_config_from_reseller(api_binary(), ne_binary()) -> {ok, kz_json:object()} | {error, any()}.
+load_config_from_reseller(undefined, _Config) -> kz_json:new();
+load_config_from_reseller(Account, Config) ->
+    ResellerId = kz_services:find_reseller_id(Account),
+    load_config_from_account(ResellerId, Config).
+
+-spec load_config_from_system(api_binary(), ne_binary()) -> {ok, kz_json:object()} | {error, any()}.
+load_config_from_system(_Account, Config) ->
+    kz_json:get_value(<<"default">>, ensure_value(kapps_config:get_category(Config))).
+
+-spec load_default_config(api_binary(), ne_binary()) -> {ok, kz_json:object()}.
+load_default_config(_Account, Config) ->
+    Schema = system_schema(Config),
+    {ok, kz_doc:set_id(kz_json_schema:default_object(Schema), doc_id(Config))}.
+
+ensure_value({ok, JObj}) -> JObj;
```

Comment:
`esure_value/1` and `maybe_new/1` has same functionality, maybe merge it into one?